### PR TITLE
k8s_info: Add wait functionality

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

User can specify wait in k8s_info module.

Fixes: #18

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/module_utils/common.py
plugins/module_utils/raw.py
plugins/modules/k8s.py
plugins/modules/k8s_info.py

### CODE REVIEW

File: plugins/modules/k8s_info.py [hunk 1]
```diff
@@ -42,10 +42,21 @@
     description: List of field selectors to use to filter results
     type: list
     elements: str
+  state:
+    description:
+    - Desired state of the object to gather information from.
+    - Only applicable if C(wait), C(wait_sleep), C(wait_condition), and C(wait_timeout) specified.
+    - If you are waiting for object to be created by Kubernetes, set this to C(present).
+    - If you are waiting for object to be deleted by Kubernetes, set this to C(absent).
+    - This parameter is different from regular Ansible state parameter.
```
Comment (fabianvf) [inline plugins/modules/k8s_info.py:51]: Is there a convention for this or does `state` still make the most sense?

**Comment by willthames** [comment]
Do we have a use case for this change? I'm just wondering in what circumstances you'd want to wait in k8s_info rather than just waiting in k8s, or using a retry loop around k8s_info

**Comment by geerlingguy** [comment]
@willthames specifically see: https://github.com/ansible-collections/community.kubernetes/issues/18#issuecomment-682153809

**Comment by willthames** [comment]
thanks @geerlingguy. Do we ever think we'll care about waiting for things to be deleted? (I'm just keen to avoid adding `state` explicitly)

File: plugins/modules/k8s_info.py (hunk 1 above)
Comment (willthames) [reply plugins/modules/k8s_info.py:51]: I'd prefer not to add it, but have `wait_state` if we really need it

Comment (fabianvf) [reply plugins/modules/k8s_info.py:51]: yeah, I think the `absent` behavior can actually be achieved like so:

```
k8s_info:
  api_version: v1
  kind: Namespace
  name: testing
register: result
until: not result.resources
```

so really no need for `state` here IMO

Comment (Akasurde) [reply plugins/modules/k8s_info.py]: I removed `state`.

**Comment by Akasurde** [comment]
@willthames @fabianvf @geerlingguy I removed the `state` parameter and added test. Please review once. Thanks.

File: plugins/module_utils/common.py [hunk 2]
```diff
@@ -318,7 +355,9 @@ def _wait_for_elapsed():
         response = None
         while _wait_for_elapsed() < timeout:
             try:
-                response = resource.get(name=name, namespace=namespace)
+                response = resource.get(name=name, namespace=namespace,
```
Comment (fabianvf) [inline plugins/module_utils/common.py:358]: One thing to note, I think previously the implementation of `wait` dealt only with a single resource, but multiple resources can be retrieved via a selector. We either need to iterate over the returned resources here or only pass individual resource names to the `wait` function.

Comment (fabianvf) [reply plugins/module_utils/common.py:358]: We could also implement a decorator for the predicates that deals with iterating over `ResourceList` objects if they're passed in. Any way works as long as we just iterate over the individual resources at some point.

Comment (fabianvf) [reply plugins/module_utils/common.py:358]: We should also add a test for the `k8s_info` wait that handles multiple objects matching the selector (ie, two pods that share a label)

**Review by fabianvf — CHANGES_REQUESTED** [review]
One spot I noticed a potential issue

**Pull request closed without merging**